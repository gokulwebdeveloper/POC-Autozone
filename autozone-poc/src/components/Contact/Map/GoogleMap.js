import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { Map, GoogleApiWrapper } from "google-maps-react";

const mapStyles = {
  width: "100%",
  height: "100%",
};

function GoogleMap(props) {
  return (
    <Map
      google={props.google}
      zoom={props.zoom}
      style={mapStyles}
      initialCenter={props.initialCenter}
    />
  );
}

GoogleMap.propTypes = {
  initialCenter: PropTypes.object,
  zoom: PropTypes.number,
}

GoogleMap.defaultProps = {
  initialCenter: {
    lat: 28.527582,
    lng: 77.0688991,
  },
  zoom: 14,
};

export default GoogleApiWrapper({
  apiKey: process.env.google_maps_api_key,
})(GoogleMap);
