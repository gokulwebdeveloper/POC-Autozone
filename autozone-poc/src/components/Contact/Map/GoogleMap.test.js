import React from "react";
import { mount, shallow } from "enzyme";
import renderer from "react-test-renderer";
import GoogleMap from "./GoogleMap";

let wrapper;
//mount component for testing
beforeEach(() => {
  wrapper = shallow(<GoogleMap />);
});

//Cleanup after test execution
afterEach(() => {
  wrapper.unmount();
});

describe("test CheckoutPage", () => {
  //test if CheckoutPage component exists
  it("CheckoutPage Exists", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
  //SnapShot Match test
  it("SnapShot Match", () => {
    const wrapper = renderer.create(<GoogleMap />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
