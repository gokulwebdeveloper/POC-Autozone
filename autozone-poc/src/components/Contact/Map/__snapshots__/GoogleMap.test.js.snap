// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CheckoutPage SnapShot Match 1`] = `
<div>
  Loading...
</div>
`;

exports[`test GoogleMap SnapShot Match 1`] = `
<div>
  Loading...
</div>
`;
